pkg/controller/node/nodecontroller.go

```diff
@@ -220,6 +243,12 @@ func NewNodeController(
 		controller.NoResyncPeriodFunc(),
 		framework.ResourceEventHandlerFuncs{},
 	)
+
+	if allocateNodeCIDRs {
+		// TODO(mqliang): make pod CIDR mask size configurable, for now set it to 24.
+		nc.cidrAllocator = NewCIDRRangeAllocator(clusterCIDR, 24)
```

Comment:
You need to go through existing Nodes and populate 'usedCIDR' map. Otherwise this won't work after controllerManager restart.